src/sentry/monitors/tasks.py

```diff
@@ -390,3 +407,51 @@ def mark_checkin_timeout(checkin_id: int, ts: datetime, **kwargs):
         )
 
         mark_failed(checkin, ts=most_recent_expected_ts)
+
+
+@instrumented_task(
+    name="sentry.monitors.tasks.detect_broken_monitor_envs",
+    max_retries=0,
+    time_limit=15 * 60,
+    soft_time_limit=10 * 60,
+    record_timing=True,
+)
+def detect_broken_monitor_envs():
+    current_time = django_timezone.now()
+    open_incidents = MonitorIncident.objects.select_related("monitor").filter(
```

Comment:
We could potentially fetch a large number of rows here and OOM. To avoid this, we can do 
https://github.com/getsentry/sentry/blob/51cbe95d93915e4111486997e8c71c9274dfbaa5/src/sentry/tasks/auto_ongoing_issues.py#L136-L137

And process a large chunk at a time.

Reply:
@davidenwang should we also go ahead and join the MonitorEnvBrokenDetection table here and filter out incidents that have already been notified on?

Reply:
hmm yea that makes sense, I guess looking down the road eventually we will want to take action (muting/disabling) on even the incidents that have received notification. Though I guess in that case I should've added a muted_or_disabled timestamp to the MonitorEnvBrokenDetection table and then i can filter off that. I can definitely add those as we go though, since right now filtering off that won't change anything until we add notifications and automated muted/disabling

Reply:
gonna add a TODO here for future